app/src/main/java/com/raulh82vlc/functionalkotlinarrow/ui/adapter/FeedAdapter.kt

```diff
@@ -21,14 +21,14 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import com.raulh82vlc.functionalkotlinarrow.R
-import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCacheModel
+import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCache
 import kotlinx.android.extensions.LayoutContainer
 import kotlinx.android.synthetic.main.item_list.view.*
 import com.squareup.picasso.Picasso.with as withPicasso
 
 class FeedAdapter(
-        var feedList: List<FeedItemCacheModel> = ArrayList(),
-        val itemClick: (FeedItemCacheModel) -> Unit) : RecyclerView.Adapter<FeedAdapter.FeedViewHolder>() {
+        var feedList: List<FeedItemCache> = ArrayList(),
```

Comment:
If you move feedList as a property in the body with delegation:
```
var values : List<FeedItemCacheModel> = emptyList()
    set(value) { notifyDataSetChanged() }
}
```
Then you save having to notify from outside when modifying the values. Also, I would avoid using `ArrayList` and go for `listOf()` or `emptyList()`, specially since in this case you want an empty list which is constant: https://github.com/JetBrains/kotlin/blob/1.2.50/libraries/stdlib/src/kotlin/collections/Collections.kt#L70